sound/soc/sof/ipc4-topology.c

```diff
@@ -819,6 +837,133 @@ static void sof_ipc4_unprepare_copier_module(struct snd_sof_widget *swidget)
 	ida_free(&fw_module->m_ida, swidget->instance_id);
 }
 
+static int snd_sof_get_pcm_hw_params(struct snd_sof_dev *sdev, struct snd_pcm_hw_params *params,
+				     int *sample_rate, int *channel_count, int *bit_depth)
+{
+	int depth;
+
+	switch (params_format(params)) {
+	case SNDRV_PCM_FORMAT_S16_LE:
+		depth = 16;
+		break;
+	case SNDRV_PCM_FORMAT_S24_LE:
+		depth = 24;
+		break;
+	case SNDRV_PCM_FORMAT_S32_LE:
+		depth = 32;
+		break;
+	default:
+		dev_err(sdev->dev, "invalid pcm frame format %d\n", params_format(params));
+		return -EINVAL;
+	}
+
+	*bit_depth = depth;
+	*channel_count = params_channels(params);
+	*sample_rate = params_rate(params);
+
+	dev_dbg(sdev->dev, "%s: rate %d channels %d sample_valid_bytes %d\n",
```

Comment:
sample_valid_bytes is in bits...

I would use different order: rate, sample_width, channels
and probably in a form of 48000Hz, 24bits, 2 channels (<- but it is likely only me).